core/src/main/java/bisq/core/app/BisqSetup.java

```diff
@@ -729,9 +731,65 @@ private void maybeShowSecurityRecommendation() {
+        maybeTriggerDisplayHandler("bitcoinLocalhostNode", displayLocalhostHandler, bisqEnvironment.isBitcoinLocalhostNodeRunning());
+    }
+
+    private void maybeShowAccountSigningStateInfo() {
+        String keySignedByArbitrator = "accountSignedByArbitrator";
+        String keySignedByPeer = "accountSignedByPeer";
+        String keyPeerLimitedLifted = "accountLimitLifted";
+        String keyPeerSigner = "accountPeerSigner";
+
+        // check signed witness on startup
+        checkSigningState(AccountAgeWitnessService.SignState.ARBITRATOR, keySignedByArbitrator, displaySignedByArbitratorHandler);
+        checkSigningState(AccountAgeWitnessService.SignState.PEER_INITIAL, keySignedByPeer, displaySignedByPeerHandler);
+        checkSigningState(AccountAgeWitnessService.SignState.PEER_LIMIT_LIFTED, keyPeerLimitedLifted, displayPeerLimitLiftedHandler);
+        checkSigningState(AccountAgeWitnessService.SignState.PEER_SIGNER, keyPeerSigner, displayPeerSignerHandler);
+
+        // check signed witness during runtime
+        p2PService.getP2PDataStorage().addAppendOnlyDataStoreListener(
+                payload -> {
+                    maybeTriggerDisplayHandler(keySignedByArbitrator, displaySignedByArbitratorHandler,
+                            isSignedWitnessOfMineWithState(payload, AccountAgeWitnessService.SignState.ARBITRATOR));
+                    maybeTriggerDisplayHandler(keySignedByPeer, displaySignedByPeerHandler,
+                            isSignedWitnessOfMineWithState(payload, AccountAgeWitnessService.SignState.PEER_INITIAL));
+                    maybeTriggerDisplayHandler(keyPeerLimitedLifted, displayPeerLimitLiftedHandler,
+                            isSignedWitnessOfMineWithState(payload, AccountAgeWitnessService.SignState.PEER_LIMIT_LIFTED));
+                    maybeTriggerDisplayHandler(keyPeerSigner, displayPeerSignerHandler,
+                            isSignedWitnessOfMineWithState(payload, AccountAgeWitnessService.SignState.PEER_SIGNER));
+                });
+    }
+
+    private void checkSigningState(AccountAgeWitnessService.SignState state,
+                                   String key, Consumer<String> displayHandler) {
+        boolean signingStateFound = p2PService.getP2PDataStorage().getAppendOnlyDataStoreMap().values().stream()
+                .anyMatch(payload -> isSignedWitnessOfMineWithState(payload, state));
+
+        maybeTriggerDisplayHandler(key, displayHandler, signingStateFound);
+    }
+
+    private boolean isSignedWitnessOfMineWithState(PersistableNetworkPayload payload,
+                                                   AccountAgeWitnessService.SignState state) {
+        if (payload instanceof SignedWitness && user.getPaymentAccounts() != null) {
+            // We know at this point that it is already added to the signed witness list
+            // Check if new signed witness is for one of my own accounts
+
+            return user.getPaymentAccounts().stream()
```

Comment:
I find this quite odd. First checking if the payload is a SignedWitness and then check if the user has a SignedWitness with that state. It's not actually checking if the payload witness has the desired SignState.